src/libclipper/include/clipper/containers.hpp

```diff
@@ -31,13 +32,14 @@ class ModelContainer {
 
   size_t get_batch_size(Deadline deadline);
   double get_average_throughput_per_millisecond();
-  void update_throughput(size_t batch_size, long total_latency);
+  void update_container_stats(size_t batch_size, long total_latency);
```

Comment:
I'd rather we separate the critical task of updating container throughput for proper batching from the helpful but non-critical metrics update. Let's call `processing_container->latency_hist_.insert(total_latency)` after `processing_container->update_throughput` within the task executor.